lib/opte/src/engine/dhcpv6/protocol.rs

```diff
@@ -363,10 +370,35 @@ fn generate_reply_options<'a>(
     // resolve them.
     if (msg.has_option(OptionCode::DomainList)
         || msg.has_option_request_with(OptionCode::DomainList))
-        && !action.domain_list.is_empty()
+        && !dhcp_state.domain_search_list.is_empty()
     {
-        let opt = Dhcpv6Option::from(action.domain_list.as_slice());
-        options.push(opt);
+        // let opt = Dhcpv6Option::DomainList(dhcp_state.domain_search_list.clone().into());
+        let opt = Dhcpv6Option::from(dhcp_state.domain_search_list.as_slice());
+
+        // Slightly hacky assertion that the contents are owned.
+        let Dhcpv6Option::DomainList(Cow::Owned(raw_list)) = opt else {
+            panic!("DHCPv6 DomainList creation allocs into new vec -- not found?")
+        };
+
+        options.push(Dhcpv6Option::DomainList(Cow::Owned(raw_list)));
+    }
+
+    if msg.has_option(OptionCode::Fqdn)
+        || msg.has_option_request_with(OptionCode::Fqdn)
+    {
+        // XXX: We should verify customer flow here -- correct
+        //      for internal DNS, maybe not external?
+        // Flags: we are (O)verriding client preference, and (S)erver is
+        // responsible for installing DNS records.
+        //                   xxxx_xNOS
```

Comment:
Mention what the N flag is and why we're clearing it?

Reply:
[RFC 4704 § 4.1] covers the flags field:

```
   The format of the Flags field is:

        0 1 2 3 4 5 6 7
       +-+-+-+-+-+-+-+-+
       |  MBZ    |N|O|S|
       +-+-+-+-+-+-+-+-+


   The "S" bit indicates whether the server SHOULD or SHOULD NOT perform
   the AAAA RR (FQDN-to-address) DNS updates.  A client sets the bit to
   0 to indicate that the server SHOULD NOT perform the updates and 1 to
   indicate that the server SHOULD perform the updates.  The state of
   the bit in the reply from the server indicates the action to be taken
   by the server; if it is 1, the server has taken responsibility for
   AAAA RR updates for the FQDN.

   The "O" bit indicates whether the server has overridden the client's
   preference for the "S" bit.  A client MUST set this bit to 0.  A
   server MUST set this bit to 1 if the "S" bit in its reply to the
   client does not match the "S" bit received from the client.

   The "N" bit indicates whether the server SHOULD NOT perform any DNS
   updates.  A client sets this bit to 0 to request that the server
   SHOULD perform updates (the PTR RR and possibly the AAAA RR based on
   the "S" bit) or to 1 to request that the server SHOULD NOT perform
   any DNS updates.  A server sets the "N" bit to indicate whether the
   server SHALL (0) or SHALL NOT (1) perform DNS updates.  If the "N"
   bit is 1, the "S" bit MUST be 0.

   The remaining bits in the Flags field are reserved for future
   assignment.  DHCPv6 clients and servers that send the Client FQDN
   option MUST clear the MBZ bits, and they MUST ignore these bits.
```

So I guess in the response, `S` is "whether the server is installing AAAA resource records", and `N` is "whether the server is installing _any_ resource record changes (AAAA, PTR, or whatever shows up in future RFCs)". I'll add a note about what N is.